xray/services/summary.go

```diff
@@ -53,6 +55,59 @@ func (ss *SummaryService) GetBuildSummary(params XrayBuildParams) (*SummaryRespo
 	return &summaryResponse, nil
 }
 
+func (ss *SummaryService) GetArtifactSummary(params ArtifactSummaryParams) (*ArtifactSummaryResponse, error) {
+	httpDetails := ss.XrayDetails.CreateHttpClientDetails()
+	servicesutils.SetContentType("application/json", &httpDetails.Headers)
+
+	requestBody, err := json.Marshal(params)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+
+	url := fmt.Sprintf("%sartifact", ss.getSummeryUrl())
+	resp, body, err := ss.client.SendPost(url, requestBody, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+	if err = errorutils.CheckResponseStatus(resp, http.StatusOK); err != nil {
+		return nil, errorutils.CheckError(errorutils.GenerateResponseError(resp.Status, utils.IndentJson(body)))
+	}
+	var response ArtifactSummaryResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	if response.Errors != nil && len(response.Errors) > 0 {
+		return nil, errorutils.CheckErrorf("Getting artifact-summery for artifact: %s failed with error: %s", response.Errors[0].Identifier, response.Errors[0].Error)
+	}
+	return &response, nil
+}
+
+type ArtifactSummaryParams struct {
+	Checksums []string `json:"checksums,omitempty"`
+	Paths     []string `json:"paths,omitempty"`
+}
+
+type ArtifactSummaryResponse struct {
+	Artifacts []Artifact `json:"artifacts,omitempty"`
+	Errors    []Error    `json:"errors,omitempty"`
+}
+
+type Artifact struct {
+	General General `json:"general,omitempty"`
+	Issues  []Issue `json:"issues,omitempty"`
+	// TODO: Create License struct with correct fields for api/v2/summary/artifact endpoint
+	// Licenses []License `json:"licenses,omitempty"`
```

Comment:
A `License` struct already exists https://github.com/jfrog/jfrog-client-go/blob/e3c6206b724288677c125d89268f461a78c33e57/xray/services/scan.go#L225-L231, but does not match the api specifiaction of https://www.jfrog.com/confluence/display/XRAY2X/Xray+REST+API#XrayRESTAPI-ArtifactSummary .

Is the existing one correct, and should I simply create a new `ArtifactLicense` struct?

Reply:
The existing [`License` struct](https://github.com/jfrog/jfrog-client-go/blob/e3c6206b724288677c125d89268f461a78c33e57/xray/services/scan.go#L225-L231) belongs to the `/scan/graph` API it seems, when comparing to the TypeScript library and Java library they have created two different datatypes for the license used in scan graph summary and in summary:

*TypeScript*:
* [Scan/GraphLicense.ts](https://github.com/jfrog/jfrog-client-js/blob/master/model/Xray/Scan/GraphLicense.ts)
* [Summary/License.ts](https://github.com/jfrog/jfrog-client-js/blob/master/model/Xray/Summary/License.ts)

*Java*:
* [scan/License.java](https://github.com/jfrog/xray-client-java/blob/master/src/main/java/com/jfrog/xray/client/services/scan/License.java)
* [summary/License.java](https://github.com/jfrog/xray-client-java/blob/master/src/main/java/com/jfrog/xray/client/services/summary/License.java)

So I think it is safe to say that you should create a new license struct.

Reply:
Right - please go ahead and create it. We rally prefer not to merge PRs with // TODO comments or commented out code @mikaello .